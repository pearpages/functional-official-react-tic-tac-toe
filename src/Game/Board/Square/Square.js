import React from 'react'
import PropTypes from 'prop-types'

import './Square.css'

export function Square({onClick, value, isHighlighted}) {
  return (
    <button className={`square ${isHighlighted ? Square.HIGHLIGHTED : ''}`} onClick={onClick}>
      {value}
    </button>
  );
}

Square.HIGHLIGHTED = 'square--highlighted'

Square.renderSquare = ({value = '', clickHandler = () => undefined, isHighlighted = false}) => {
  return <Square
    value={value}
    onClick={clickHandler}
    isHighlighted={isHighlighted}
  />;
}

Square.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string,
  isHighlighted: PropTypes.bool
}
