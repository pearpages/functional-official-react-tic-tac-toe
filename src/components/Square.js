import React from 'react'

export function Square({onClick, value}) {
  return (
    <button className="square" onClick={onClick}>
      {value}
    </button>
  );
}

Square.renderSquare = ({value, clickHandler}) => {
  return <Square 
    value={value} 
    onClick={clickHandler}
  />;
}
