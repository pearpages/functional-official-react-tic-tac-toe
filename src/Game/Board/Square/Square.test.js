import { render, fireEvent } from '@testing-library/react'

import { Square } from './Square'

describe("<Square>", () => {
  it("Should render a button with the value, by default does not containt square--highlighted as a className", () => {
    const value = "My text"
    const result = render(<Square onClick={() => undefined} value={value}/>)

    const button = result.getByRole('button')
    expect(button.textContent).toEqual(value)

    expect(result.container.getElementsByClassName(Square.HIGHLIGHTED).length).toBe(0)
    expect(result.container.getElementsByClassName('square').length).toBe(1)
  })

  it("Should call the onClick function when the button gets clicked", () => {
    const value = "My text"
    const onClick = jest.fn()
    const result = render(<Square onClick={onClick} value={value}/>)

    const button = result.getByRole('button')
    fireEvent.click(button)
    expect(onClick).toHaveBeenCalled()
  })
})

describe("renderSquare", () => {
  test('renders a Square component with all the params set', () => {
    const value = 'My other text'
    const clickHandler = jest.fn()
    const isHighlighted = true
    const result = render(Square.renderSquare({
      value,
      clickHandler,
      isHighlighted
    }))

    expect(result.container.textContent).toEqual(value)
    expect(result.container.getElementsByClassName(Square.HIGHLIGHTED).length).toBe(1)
  })
})