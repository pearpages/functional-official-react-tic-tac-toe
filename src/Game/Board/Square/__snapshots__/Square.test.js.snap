// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square> When Squares snapshot changes 1`] = `"<button class=\\"square \\">My text</button>"`;
